runtime/lua/vim/lsp/completion.lua

```diff
@@ -471,6 +476,222 @@ local function request(clients, bufnr, win, ctx, callback)
+--- @param param table The parameters for the LSP request
+--- @param selected_word string The word being completed
+function CompletionResolver:request(bufnr, param, selected_word)
+  self:cleanup()
+
+  self.bufnr = bufnr
+  self.word = selected_word
+  local debounce_time = self:next_debounce_time()
+
+  -- Create and start the timer
+  self.timer = assert(vim.uv.new_timer())
+  self.timer:start(
+    debounce_time,
+    0,
+    vim.schedule_wrap(function()
+      local valid, cmp_info = self:is_valid()
+      if not valid then
+        self:cleanup()
+        return
+      end
+
+      -- Cancel any pending requests before sending a new one
+      self:cancel_pending_requests()
+
+      local client_id = vim.tbl_get(cmp_info.completed, 'user_data', 'nvim', 'lsp', 'client_id')
+      local client = client_id and vim.lsp.get_client_by_id(client_id)
+      if not client then
+        return
+      end
+
+      local start_time = vim.uv.hrtime()
+      self.last_request_time = start_time
+
+      local ok, request_id = client:request(
+        vim.lsp.protocol.Methods.completionItem_resolve,
+        param,
+        function(err, result)
+          local end_time = vim.uv.hrtime()
+          local response_time = (end_time - start_time) * ns_to_ms
+          self.doc_rtt_ms = self.doc_compute_new_average(response_time)
+
+          if err or not result or next(result) == nil then
+            if err then
+              vim.notify(err.message, vim.log.levels.WARN)
+            end
+            return
+          end
+
+          valid, cmp_info = self:is_valid()
+          if not valid then
+            return
+          end
+
+          local value = vim.tbl_get(result, 'documentation', 'value')
+          if not value then
+            return
+          end
+          local windata = vim.api.nvim__complete_set(cmp_info.selected, {
+            info = value,
+          })
+          local kind = vim.tbl_get(result, 'documentation', 'kind')
+          update_popup_window(windata.winid, windata.bufnr, kind)
+        end,
+        bufnr
+      )
+
+      if ok and request_id then
+        self.request_ids[client.id] = request_id
+      end
+    end)
+  )
+end
+
+--- Defines a CompleteChanged handler to request and display LSP completion item documentation
+--- via completionItem/resolve
+local function on_completechanged(group, bufnr)
+  api.nvim_create_autocmd('CompleteChanged', {
+    group = group,
+    buffer = bufnr,
+    callback = function(args)
+      local completed_item = vim.v.event.completed_item or {}
+      if (completed_item.info or '') ~= '' then
+        local data = vim.fn.complete_info({ 'selected' })
+        update_popup_window(data.preview_winid, data.preview_bufnr)
+        return
+      end
+
+      if
+        #lsp.get_clients({
+          id = vim.tbl_get(completed_item, 'user_data', 'nvim', 'lsp', 'client_id'),
+          method = ms.completionItem_resolve,
```

Comment:
```suggestion
          method = 'completionItem/resolve',
```
Similarly: deprecations + unavailable `ms`.